Tests/ArcGISToolkitTests/PreplannedMapModelTests.swift

```diff
@@ -87,7 +87,7 @@ class PreplannedMapModelTests: XCTestCase {
     
     @MainActor
     func testPackagingStatus() async throws {
-        class MockPreplannedMapArea: PreplannedMapAreaProtocol {
+        final class MockPreplannedMapArea: PreplannedMapAreaProtocol, @unchecked Sendable {
```

Comment:
While this works, it would be better for the sendability to be checked. You could do that by importing `os` and then using `OSAllocatedUnfairLock`:

```swift
final class MockPreplannedMapArea: PreplannedMapAreaProtocol {
    let packagingStatus = OSAllocatedUnfairLock<PreplannedMapArea.PackagingStatus?>(initialState: nil)
    
    func retryLoad() async throws {
        packagingStatus.withLock { $0 = .processing }
    }
}
```

Similar with the other mocks that need to set the packaging status.

Reply:
I tried to apply this feedback and the status initialization fails on this line, so the test assertion fails:
```
status = preplannedMapArea.packagingStatus.map(Status.init) ?? .packaged
```
```
model.status.assertExpectedValue(.packaging) // assertion fails because status equals `.packaged`
```
With the current implementation the status is set with the initializer successfully so the test succeeds. I did not apply this feedback for now. Does the assertion need to be changed?

Reply:
I'm a bit confused, but I'll re-review the PR and comment further based on what I see.

Reply:
Ah! I see. I'm sorry, I should have actually run the test before making my suggestion. The problem is that `packagingStatus` no longer satisfied the protocol requirement. Here is how to fix that:

```swift
final class MockPreplannedMapArea: PreplannedMapAreaProtocol {
    var packagingStatus: PreplannedMapArea.PackagingStatus? {
        _packagingStatus.withLock { $0 }
    }
    
    private let _packagingStatus = OSAllocatedUnfairLock<PreplannedMapArea.PackagingStatus?>(initialState: nil)
    
    func retryLoad() async throws {
        _packagingStatus.withLock { $0 = .processing }
    }
}
```